### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by nicolercl**
Add log in & log out functions. Visitors need to log in to leave comments.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.getWriter().println("Hello " + userEmail + ".");
+      response.getWriter().println("Leave a comment below or log out <a href=\"" + logoutUrl + 
+      "\">here.</a>");
+
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
```
Comment (shadyof) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Curious that could line 22, 23 and 30,31 be refactor as a method?

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
```
Comment (shadyof) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: nit: extra line?

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+@WebServlet("/")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+    
+
+    UserService userService = UserServiceFactory.getUserService();
```
Comment (shadyof) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: nit: final
If this object used very often then maybe consider make it as a member.

**Review by shadyof — COMMENTED** [review]
leave some comments and take my lunch first

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java (hunk 1 above)
Comment (nicolercl) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: 22, 23 uses the createLoginURL() function while 30, 31 uses the createLogoutURL(), so I don't think that's possible.

Comment (shadyof) [reply portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Ah okay thanks!

**Review by shadyof — APPROVED** [review]

**Pull request merged**